client/src/App.jsx

```diff
@@ -1,152 +1,186 @@
-                    isCorrect: isNameCorrect,
-                }
-            } else {
-                // User did not enter a name for this person.
-                return {actualUser, isCorrect: false}
-            }
-        })
-        setEnteredNames(namesValidated);
-        setIsGameOver(true)
-    }
-
-    function handleNameEntered(name, id) {
-        setEnteredNames((prevEnteredNames) => {
-            // Check if the user with the same ID already exists in the array.
-            // If the user exists, update their name, else add a new user to the array.
-            const userIndex = prevEnteredNames.findIndex((user) => user.id === id);
-            if (userIndex !== -1) {
-                prevEnteredNames[userIndex].name = name;
-            } else {
-                prevEnteredNames.push({id: id, name: name});
-            }
-            return [...prevEnteredNames];
-        });
-    }
-
-    function getScore() {
-        const correctCount = enteredNames.filter((user) => user.isCorrect).length
-        return correctCount
-    }
-
-
-    return (
-        <main>
-            {!isGameStarted && <StartMenu onStartGame={handleStartGame}/>}
-            {isWaitingTestStart ? <TestCountdown handleTestCountdown={handleTestCountdown}/> :
-                <div className="user-cards-container">
-                    {randomUserElements}
-                </div>}
-            {isLearningPhase &&
-                <p className="test-countdown-container"> {learningPhaseTimeRemainingInSeconds} seconds remaining</p>
-            }
-            {isLearningPhase && <button className="test-button" onClick={handleTestStart}>Test</button>}
-            {isGameStarted && !isLearningPhase && !isWaitingTestStart && !isGameOver &&
-                <button className="submit-button" onClick={handleTestSubmit}>Finish Test</button>
-            }
-            {isGameOver && (
-                <React.Fragment>
-                    <ScoreMessage correctAnswersCount={getScore()} totalUsers={randomUsers.length}/>
-                    <button className="restart-button" onClick={handleGameRestart}>Restart Test</button>
-
-                </React.Fragment>
-            )
-            }
-        </main>
-    )
-}
\ No newline at end of file
+  const [randomUsers, setRandomUsers] = React.useState([]);
+  const [isGameStarted, setIsGameStarted] = React.useState(false);
+  const [isLearningPhase, setIsLearningPhase] = React.useState(false);
+  const [isWaitingTestStart, setIsWaitingTestStart] = React.useState(false);
+  const [isGameOver, setIsGameOver] = React.useState(false);
+  const [enteredNames, setEnteredNames] = React.useState([]);
+  const [
+    learningPhaseTimeRemainingInSeconds,
+    setLearningPhaseTimeRemainingInSeconds,
+  ] = React.useState(LEARNING_PHASE_DURATION_IN_SECONDS);
+  //   const [isStartGameInstruction, setIsStartGameInstruction] = useState(false);
+  const [showInstructions, setShowInstructions] = React.useState(false);
```

Comment:
showInstructions should be kept in the start menu component. the app component doesnt need to know about the state of the instructions. this will work but its unnecessary